blaze/expr/expressions.py

```diff
@@ -713,7 +713,7 @@ def apply(expr, func, dshape, splittable=False):
     return Apply(expr, func, datashape.dshape(dshape), splittable)
 
 
-class Coerce(Expr):
+class Coerce(ElemWise):
```

Comment:
I like the practicality of this; however, I wonder if it would make more sense to make the syntax for this something like:

```python
arr = bz.Data([1, 2, 3])
asfloat = arr.coerce('var * float64')
```

This might even let us do some fun things with reshaping like:

```python
arr = bz.Data([1, 2, 3, 4])
asmat = arr.coerce('2 * 2 * int32')
```

Just a thought, not sure if this is practical.